src/MongoDB/Driver/ReadPreference.cpp

```diff
@@ -69,12 +69,24 @@ Array HHVM_METHOD(MongoDBDriverReadPreference, __debugInfo)
 	Variant v_tags;
 	const bson_t *tags = mongoc_read_prefs_get_tags(data->m_read_preference);
 
-	retval.set(s_mode, mongoc_read_prefs_get_mode(data->m_read_preference));
+	mongoc_read_mode_t mode = mongoc_read_prefs_get_mode(data->m_read_preference);
+
+	switch (mode) {
+		case MONGOC_READ_PRIMARY: retval.set(s_mode, "primary"); break;
+		case MONGOC_READ_PRIMARY_PREFERRED: retval.set(s_mode, "primaryPreferred"); break;
+		case MONGOC_READ_SECONDARY: retval.set(s_mode, "secondary"); break;
+		case MONGOC_READ_SECONDARY_PREFERRED: retval.set(s_mode, "secondaryPreferred"); break;
+		case MONGOC_READ_NEAREST: retval.set(s_mode, "nearest"); break;
+		default: /* Do nothing */
+			break;
```

Comment:
FWIW, libmongoc defaults to an empty string (see: [here](https://github.com/mongodb/mongo-c-driver/blob/7f6ae5b5f36fb55959e8137efb5c07fef472e26a/src/mongoc/mongoc-read-prefs.c#L249)) for an unknown value. That said, we wouldn't allow construction of a ReadPreference with an unsupported mode and `mongoc-uri.c` defaults to a primary read preference, so this edge case should never be reached.